test/e2e/telemetry_test.go

```diff
@@ -100,13 +103,78 @@ var _ = Describe("Telemetry-module", func() {
 			})
 
 			Eventually(func(g Gomega) {
-				Eventually(func(g Gomega) {
-					var secret corev1.Secret
-					g.Expect(k8sClient.Get(ctx, webhookCertSecret, &secret)).Should(Succeed())
-					g.Expect(secret.OwnerReferences).Should(HaveLen(1))
-					g.Expect(secret.UID).ShouldNot(Equal(oldUID))
-				}, timeout, interval).Should(Succeed())
+				var secret corev1.Secret
+				g.Expect(k8sClient.Get(ctx, webhookCertSecret, &secret)).Should(Succeed())
+				g.Expect(secret.OwnerReferences).Should(HaveLen(1))
+				g.Expect(secret.UID).ShouldNot(Equal(oldUID))
+			}, timeout, interval).Should(Succeed())
+		})
+	})
+
+	Context("Deleting telemetry resources", Ordered, func() {
+		telemetryKey := types.NamespacedName{
+			Name: "default",
+		}
+		k8sLogPipelineObject := makeTestPipelineK8sObjects()
+
+		BeforeAll(func() {
+			Eventually(func(g Gomega) {
+				g.Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sLogPipelineObject...)).Should(Succeed())
+			}, timeout, interval).Should(Succeed())
+		})
+
+		AfterAll(func() {
+			// Re-create telemetry to have ValidatingWebhookConfiguration for remaining tests
+			Eventually(func(g Gomega) {
+				newTelemetry := []client.Object{kitk8s.NewTelemetry("default").K8sObject()}
+				g.Expect(kitk8s.CreateObjects(ctx, k8sClient, newTelemetry...)).Should(Succeed())
+			}, timeout, interval).Should(Succeed())
+
+			Eventually(func(g Gomega) {
+				var telemetry v1alpha1.Telemetry
+				g.Expect(k8sClient.Get(ctx, telemetryKey, &telemetry)).Should(Succeed())
+				g.Expect(telemetry.Status.State).Should(Equal(v1alpha1.StateReady))
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should have telemetry resource", func() {
+			Eventually(func(g Gomega) {
+				var telemetry v1alpha1.Telemetry
+				g.Expect(k8sClient.Get(ctx, telemetryKey, &telemetry)).Should(Succeed())
+				g.Expect(telemetry.Status.State).Should(Equal(v1alpha1.StateReady))
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should not delete telemetry when LogPipeline exists", func() {
+			By("Deleting telemetry", func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, telemetryK8sObjects...)).Should(Succeed())
+			})
+
+			Eventually(func(g Gomega) {
+				var telemetry v1alpha1.Telemetry
+				g.Expect(k8sClient.Get(ctx, telemetryKey, &telemetry)).Should(Succeed())
+				g.Expect(telemetry.Status.State).Should(Equal(v1alpha1.StateError))
+				g.Expect(telemetry.Finalizers).Should(HaveLen(1))
+				g.Expect(telemetry.Finalizers[0]).Should(Equal("telemetry.kyma-project.io/finalizer"))
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should delete telemetry", func() {
```

Comment:
Let's capitalize Telemetry in test case description to stress out that the CR is meant

Reply:
Done